### DISCUSSION TITLE
Cannot resolve injected property names

### DISCUSSION THREAD

**Discussion by rtabulov** [vuejs/language-tools#437] [Q&A] [answered]
in vscode i keep getting `Cannot find name '$t'.`

`$t` comes from vue-i18n, and is actually defined. 
i18n options - ` { legacy: false, globalInjection: true, ... }`

any suggestion on how to remove this annoying error message?

**Comment by johnsoncodehk** [comment]
Will check this, could you provide reproduce by https://github.com/johnsoncodehk/volar-starter?

**Comment by rtabulov** [comment, marked as answer]
hi, i found out that the error is coming from vscode's 'ts language features'. 
it only appears when the file only has the `<template>` part and does not have a `<script>` part.

when i am using vetur this error does not appear 🤷

i will try to reproduce it with volar-starter

attaching a screenshot to make it more clear

![image](https://user-images.githubusercontent.com/29009741/131449133-6dfa7302-1988-4a45-a10f-da45ead0c9bf.png)

**Reply by johnsoncodehk** [reply]
Fixed in https://github.com/johnsoncodehk/volar/commit/fdb03731cad2c1c9edc34b05fcee524c688fa3c2, thank you!

**Reply by rtabulov** [reply]
thanks to you! 🎉🎉

**Comment by ghost** [comment]
![2021-08-30_093834](https://user-images.githubusercontent.com/11320080/131499090-4c7c7313-8571-4885-a232-e210f69ed418.png)

Adding `script` section helps but in a weird way:
With empty `<script></script>` it shows `any` on hover
With empty `<script setup></script>` it shows `var $route: RouteLocationNormalizedLoaded`

**Reply by johnsoncodehk** [reply]
This is expected, because empty `<script setup></script>` ~= `<script>export default defineComponent({})</script>`.
